hs_core/hydroshare/utils.py

```diff
@@ -433,11 +433,19 @@ def copy_resource_files_and_AVUs(src_res_id, dest_res_id, set_to_private=False):
     tgt_coll = tgt_res.root_path
     for avu_name in avu_list:
         value = istorage.getAVU(src_coll, avu_name)
+        # possibly create private copy if requested
```

Comment:
Probably. This is more of a quick fix than a comprehensive solution. 
(The code as-is is buggy.)

We can put that in an issue but I am shooting for a working solution in 1.10.0 right now, and I **know** the existing code is wrong.